akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -130,6 +130,7 @@ private[projection] abstract class InternalProjectionState[Offset, Envelope](
           Flow[ProjectionContextImpl[Offset, Envelope]].mapAsync(parallelism = 1) { context =>
             val measured: () => Future[Done] = { () =>
               handler.process(context.envelope).map { done =>
+                statusObserver.afterProcess(projectionId, context.envelope)
                 // `telemetry.afterProcess` is invoked immediately after `handler.process`
                 telemetry.afterProcess(context.externalContext)
                 done
```

Comment:
changes in `InternalProjectionState` are the key improvement in this PR. 

An important detail is that `statusObserver.afterProcess` is invoked in the same future than `telemetry.afterProcess` and always in the same order.